src/nativeapm/nativeapm.ts

```diff
@@ -0,0 +1,469 @@
+        {
+          source: paymentConfig.gatewayConfigurationId,
+        },
+        this.handleCaptureSuccess.bind(this),
+        this.handleCaptureError.bind(this)
+      );
+    }
+
+    /**
+     * This function handles Native APM capture success
+     */
+    private handleCaptureSuccess(data: NativeApmCapturePaymentResponse) {
+      if (
+        data.success &&
+        data.native_apm &&
+        data.native_apm.state === 'PENDING_CAPTURE' &&
+        this.captureRetries < 1
+      ) {
+        this.captureRetries++;
+
+        return this.capturePayment();
+      }
+
+      if (
+        data.success &&
+        data.native_apm &&
+        data.native_apm.state === 'PENDING_CAPTURE' &&
+        this.captureRetries === 1
+      ) {
+        const errorView = new NativeApmErrorView(
+          {
+            ...this.gatewayConfiguration.native_apm,
+            errorMessage: TextUtils.getText('paymentTimeout'),
+          },
+          this.theme
+        );
+
+        EventsUtils.dispatchPaymentErrorEvent(data);
+
+        return this.loadView(errorView.getViewElement());
+      }
+
+      if (!data.success) {
+        const errorView = new NativeApmErrorView(
+          this.gatewayConfiguration.native_apm,
+          this.theme
+        );
+
+        EventsUtils.dispatchPaymentErrorEvent(data);
+
+        return this.loadView(errorView.getViewElement());
+      }
+
+      const successView = new NativeApmSuccessView(
+        this.gatewayConfiguration,
+        this.theme
+      );
+
+      EventsUtils.dispatchPaymentSuccessEvent();
+
+      return this.loadView(successView.getViewElement());
+    }
+
+    /**
+     * This function handles Native APM capture error
+     */
+    private handleCaptureError(req: XMLHttpRequest) {
+      const errorView = new NativeApmErrorView(
+        this.gatewayConfiguration.native_apm,
+        this.theme
+      );
+
+      EventsUtils.dispatchPaymentErrorEvent(req.response);
+
+      this.loadView(errorView.getViewElement());
+    }
+
+    /**
+     * This function sets theme of the Native APM widget
+     * @param {NativeApmThemeConfigType} themeConfig
+     */
+    public setTheme(themeConfig: NativeApmThemeConfigType) {
+      return this.theme.setTheme(themeConfig);
+    }
+
+    /**
+     * This function prefills user data
+     * @param {PrefilledData} data
+     */
+    public prefillData(data: PrefilledData) {
+      this.prefilledData = data;
+    }
+
+    /**
+     * This function loads view of the widget
+     */
+    private loadView(view: HTMLElement) {
+      if (this.widgetWrapper) {
+        this.napmContainer.removeChild(this.widgetWrapper);
+      }
+
+      this.widgetWrapper = this.createWidgetWrapper();
+      this.widgetWrapper.appendChild(view);
+
+      this.napmContainer.appendChild(this.widgetWrapper);
+    }
+
+    /**
+     * This function updates payment provider logo
+     */
+    private updatePaymentProviderLogo(logoUrl?: string) {
```

Comment:
This function covers case when in the response of `POST /native-payment `we get other merchant image e.g. Cellulant Mobile